### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by Saloed**
### Better support for Bv overflow check expressions
* Support overflow check expressions in all solvers by  rewriting with other Bv expressions
* Fix overflow check expressions in Bitwuzla
* Add tests

### Other improvements
* Bitwuzla: better handling of single dimension constant arrays
* Yices: allow negative indices for uninterpreted sort values
* Core: arithmetic power operation evaluator 
* Core: fp.rem operation evaluator
* Core: fp.fma operation evaluator

### CODE REVIEW

File: ksmt-bitwuzla/src/main/kotlin/io/ksmt/solver/bitwuzla/KBitwuzlaExprInternalizer.kt [hunk 1]
```diff
@@ -716,43 +724,162 @@ open class KBitwuzlaExprInternalizer(val bitwuzlaCtx: KBitwuzlaContext) : KExprL
...
+        transform(arg0, arg1) { a0: BitwuzlaTerm, a1: BitwuzlaTerm ->
+            mkBvMulSignedNoOverflowTerm(arg0.sort.sizeBits.toInt(), a0, a1, isOverflow = false)
+        }
+    }
+
+    private fun mkBvAddSignedNoOverflowTerm(
+        sizeBits: Int,
+        a0: BitwuzlaTerm,
+        a1: BitwuzlaTerm,
+        isOverflow: Boolean
```
Comment (CaelmBleidd) [inline ksmt-bitwuzla/src/main/kotlin/io/ksmt/solver/bitwuzla/KBitwuzlaExprInternalizer.kt:799]: Please, add a comment about `isOverflow` parameter

File: ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt [hunk 2]
```diff
@@ -157,12 +154,47 @@ fun <T : KArithSort> KContext.simplifyArithDiv(lhs: KExpr<T>, rhs: KExpr<T>): KE
...
+    power == RealValue.one -> base
+    power == RealValue.minusOne -> base.inverse()
+    else -> power.smallIntValue()?.let { powerIntValue ->
+        if (powerIntValue >= 0) {
+            base.pow(powerIntValue)
+        } else {
+            base.inverse().pow(powerIntValue.absoluteValue)
+        }
+    }
+}
```
Comment (CaelmBleidd) [inline ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt:190]: Can't we process NaNs here?

**Review by CaelmBleidd — APPROVED** [review]

File: ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt (hunk 2 above)
Comment (Saloed) [reply ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt:190]: Arithmetic doesn't support `NaN` special values. `NaN`s are part of the Fp theory.

Comment (CaelmBleidd) [reply ksmt-core/src/main/kotlin/io/ksmt/expr/rewrite/simplify/ArithSimplification.kt:190]: Right, my bad :)

**Pull request merged**